data/codec/src/codec_err_handler.rs

```diff
@@ -34,12 +34,15 @@ impl DecodeErrorHandler for DefaultErrorHandler {
     }
 }
 
+#[derive(Debug)]
+pub enum Never {}
```

Comment:
Turns out is already exists, we should reuse the existing one: https://doc.rust-lang.org/core/convert/enum.Infallible.html